blockchain/blockchain.go

```diff
@@ -68,7 +68,7 @@ var _ Reader = (*Blockchain)(nil)
 
 // Blockchain is responsible for keeping track of all things related to the Starknet blockchain
 type Blockchain struct {
-	network  utils.Network
+	network  utils.NetworkKnown
```

Comment:
We should keep using `utils.Network` everywhere, `utils.KnownNetwork` and  `utils.CustomNetwork` shouldn't leak beyond `utils` and `node` package context.